src/engine/strat_engine/filesystem.rs

```diff
@@ -56,6 +59,52 @@ impl StratFilesystem {
         }
     }
 
+    /// Create a snapshot of the ThinDev under the filesystem. Return the
+    /// resulting ThinDev to the caller.
+    pub fn snapshot(&self,
+                    dm: &DM,
+                    thin_pool: &ThinPoolDev,
+                    snapshot_name: &DmName,
+                    snapshot_fs_uuid: FilesystemUuid,
+                    snapshot_thin_id: ThinDevId)
+                    -> EngineResult<ThinDev> {
```

Comment:
I _think_ this should return a StratFilesystem rather than a ThinDev, with corresponding changes to ```ThinPool::snapshot_filesystem``` method.

Reply:
ok